// hardhat-live-fork/dist/src/utils/hardhat-helpers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.sendTx = exports.fundBalance = exports.impersonateAccount = void 0;
const hex_value_1 = require("./hex-value");
async function impersonateAccount(provider, address) {
    return await provider.request({
        method: "hardhat_impersonateAccount",
        params: [address],
    });
}
exports.impersonateAccount = impersonateAccount;
async function fundBalance(provider, address) {
    return await provider.request({
        method: "hardhat_setBalance",
        params: [
            address,
            "0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff",
        ],
    });
}
exports.fundBalance = fundBalance;
async function sendTx(provider, tx) {
    return await provider.request({
        method: "eth_sendTransaction",
        params: [
            {
                from: tx.from,
                to: tx.to,
                value: (0, hex_value_1.hexValue)(tx.value.toHexString()),
                data: tx.data,
                gas: (0, hex_value_1.hexValue)(tx.gasLimit.toHexString()),
                gasPrice: (0, hex_value_1.hexValue)(tx.gasPrice?.toHexString() ?? "0x0"),
            },
        ],
    });
}
exports.sendTx = sendTx;
